src/content/docs/apm/agents/java-agent/troubleshooting/classloading-issues-from-jboss-and-wildfly.mdx

```diff
@@ -0,0 +1,218 @@
+
+	In this case you need to manually configure affected JBoss Modules that are mentioned in your agent logs. For our scenario we will continue with `java.util.logging`, which is part of the `java.logging` module. Based on the stacktrace, we see the affected JBoss module is `io.undertow.servlet`. 
+
+	We can find the configuration file for a specific JBoss module in your JBoss server by: 
+
+	1. Go into the relative path `modules/system/layers/base`
+	2. Enter the file path based on the module name. In the case of `io.undertow.servlet`, this translates to `io/undertow/servlet`. 
+	3. Go into the file under the path `main/module.xml`. The full relative path for the `io.undertow.servlet` configuration file is `modules/system/layers/base/io/undertow/servlet/main/module.xml`.
+	4. In the XML file, edit the body of the `<dependencies>` tag inside the `<module>` tag. If the tag is not there, add it.
+	5. Inside the `<dependencies>` tag, add the element `<module name="$PLACEHOLDER FOR MODULE NAME"/>`. For `java.util.logging`, the element becomes `<module name="$PLACEHOLDER FOR MODULE NAME"/>`.
+
+Below is a sample XML for `io.undertow.servlet`:
+
+```xml
+<module name="io.undertow.servlet" xmlns="urn:jboss:module:1.9">
+	<resources>
+		<resource-root path="undertow-servlet-2.2.5.Final-redhat-00001.jar"/>
+	</resources>
+
+	<dependencies>
+		<module name="java.logging"/>
+
+		<module name="javax.annotation.api"/>
+		<module name="sun.jdk"/>
+		<module name="javax.servlet.api"/>
+		<module name="javax.servlet.jsp.api"/>
+		<module name="javax.servlet.jstl.api"/>
+		<module name="org.jboss.logging"/>
+		<module name="io.undertow.core"/>
+		<module name="org.jboss.xnio"/>
+		<module name="jdk.unsupported"/>
+	</dependencies>
+</module>
+```
+
+Repeat these steps for all affected modules.	
+
+    </Collapser>
+</CollapserGroup>
+
+</TabsPageItem>
+
+<TabsPageItem id="problem-application-runtime">
+
+This is the solution if your Java agent is version `8.6` or higher and you're using the Jakarta/J2EE Management API (JSR-777), but your application fails to start up. In this instance, you'll see `ClassNotFoundException`s where the package name included in the class is `javax.management.j2ee`
+
+### Solution: Fix application runtime issues [#solution-runtime]
+
+If you upgraded the Java Agent to `8.6` or higher, your application may fail to startup if you are using the Jakarta/J2EE Management API. The solution to this is to upgrade your agent to `8.7` or higher and then add the system property:
+
+	```
+	-Dcom.newrelic.jboss.jsr77.fix=true
+	```
+
+The consequence of setting the system property to true is the agent will not label `javax.management` as a system class. As a result, you'll need additional configuration. Follow these steps:
+
+	1. Edit the XML file by going to `modules/system/layers/base/io/undertow/servlet/main/module.xml`. 
```

Comment:
i notice in your original copy that you mentioned they should follow steps from another solution. I tried to adapt those steps to this content, as going to another section to troubleshoot a separate problem can be a frustrating user experience. Can you confirm that these are the steps they would take in this new context? @obenkenobi

Reply:
The steps look correct.